### ISSUE TITLE
Parsing issues on L

### ISSUE THREAD

**Issue by dhleong** [openstf/adbkit-logcat#1]
My Nexus 5 is running the L developer preview and logcat parsing does not appear to work correctly. I get a bunch of garbage, and sometimes errors about invalid format `''`.

For example, eclipse's logcat prints:

```
07-21 14:52:55.537: V/minus:instant(6496): incoming: {"type":"ONLINE_USERS","online_users":null,"live_chat_users":0,"live_chat_string":"8,764 cats hanging out..."}
```

and the JSON returned by adbkit is:

```
{
    "date": "1970-02-02T23:36:32.033Z",
    "message": "incoming: ",
    "pid": 251658326,
    "priority": 109,
    "tag": "inus:instant",
    "tid": 1666436449
}
```

Note the date, the incomplete message, the cut-off tag, etc. 

More extreme cases have large bunches of tags merged together with unicode garbage:

```
{"date":"2022-05-02T00:05:58.952Z","pid":1668246636,"tid":1667330131,"priority":115,"tag":", 112(2MB) LOS objects, 219% free, 47MB/65MB, paused 6.026ms total 53.166ms","message":"4\u0000\u0018\u0000`\u0019\u0000\u0000`\u0019\u0000\u0000�`�S���0\u0000\u0000\u0000\u0000\u0003minus::LocationUtils\u0000Using play services location!\u00009\u0000\u0018\u0000\u001d\u0003\u0000\u0000,\u0003\u0000\u0000�`�S��\u00121\u0002\u0000\u0000\u0000Nu\u0000\u0000\u0003\u0005\u0000\u0000\u0000\u0000\u0000\u0000\rY\u001d0\u0002\u0019\u0000\u0000\u0000.PluggedInAnalyzerService\u0000N'\u0000\u0000\u0000BM\u0000\u0000\n\u0010\u0000\u0018\u0000\u001d\u0003\u0000\u0000�\u0003\u0000\u0000�`�S\u001e+]1\u0002\u0000\u0000\u0000�\n\u0000\u0000\u0002\u0006\u0000\u0000\u0000Binder\n�\u0000\u0018\u0000`\u0019\u0000\u0000`\u0019\u0000\u0000�`�S\u001f��1\u0000\u0000\u0000\u0000\u0003minus::CheckInLocationListener\u0000Location changed Location[network 40.753864,-73.992259 acc=34 et=+3d20h6m51s501ms {Bundle[mParcelledData.dataSize=548]}]\u00000\u0000\u0018\u0000n\u0005\u0000\u0000�X\u0000\u0000�`�S3��1\u0000\u0000\u0000\u0000\u0004MicrophoneInputStream\u0000mic_starting eze@97e187f\u00003\u0000\u0018\u0000n\u0005\u0000\u0000�X\u0000\u0000�`�S�\r\u001a2\u0000\u0000\u0000\u0000\u0004HotwordRecognitionRnr\u0000Starting hotword detection.\u0000c\u0000\u0018\u0000`\u0019\u0000\u0000�\u001a\u0000\u0000�`�S5�\u001d2\u0000\u0000\u0000\u0000\u0003minus::CheckInLocationListener\u0000Cached location Location[fused 40.753834,-73.992279 acc=30 et=?!?]\u0000T\u0000\u0018\u0000`\u0019\u0000\u0000�\u001a\u0000\u0000�`�S�6%2\u0000\u0000\u0000\u0000\u0003minus::CheckInLocationListener\u0000Cached location time = 2014-07-21T02:44:30GMT-04:00\u00004\u0000\u0018\u0000`\u0019\u0000\u0000`\u0019\u0000\u0000�`�S?�%2\u0000\u0000\u0000\u0000\u0003minus::LocationUtils\u0000Using play services location!\u0000�\u0000\u0018\u0000`\u0019\u0000\u0000�
// continues on for a while...
```

Let me know if there's any more info I can provide. I'm getting the data with:

``` coffeescript
    adb.openLogcat(deviceId)
    .then (logcat) ->
        logcat.includeAll Logcat.Priority.VERBOSE

        console.log 'logcat!', logcat
        logcat.on 'entry', (entry) ->
            console.log 'entry', entry
```

Using my Galaxy Nexus running KitKat works just fine, which is why I suspect some binary format change on L....

**Comment by sorccu** [comment]
Unfortunately I don't have L yet - could you provide a full log file for now? Note that the log may contain sensitive information, so if that is a problem, perhaps clear the logs with `adb shell logcat -c`, then wait for a while or do something to create log events, and finally dump the log with `adb shell 'logcat -dB > /data/local/tmp/bin.log' && adb pull /data/local/tmp/bin.log`. If you'd rather not post it here, you can PM me a link to your private Dropbox, for example.

**Comment by sorccu** [comment]
Ok, I'm able to reproduce the problem with an emulator so feel free to ignore the previous request :)

**Comment by sorccu** [comment]
Ok, I've fixed the issue in the `feature/android-L` branch, but I'm getting mixed results with my emulator. Everything parses correctly, but for some reason some of the (properly formatted) entries have seemingly broken contents and priority. Could you test on your Nexus 5 and see if the same thing happens there?

**Comment by dhleong** [comment]
Thanks for the quick response! I'll check it out when I get into the office
in an hour or so.

**Comment by dhleong** [comment]
It's definitely better, but I'm also seeing the issues you mentioned.

EX:

``` javascript
{ date: Tue Jul 22 2014 10:27:38 GMT-0400 (EDT),
  pid: 24879,
  tid: 24879,
  priority: 4,
  tag: 'wpa_supplicant',
  message: 'wlan0: CTRL-EVENT-SCAN-STARTED ' }
{ date: Tue Jul 22 2014 10:27:58 GMT-0400 (EDT),
  pid: 797,
  tid: 32287,
  priority: 78,
  tag: 'u',
  message: '\u0000\u0003\u0005\u0000\u0000\u0000\u0000\u0000\u0000��\u001b\f\u0002\u0016\u0000\u0000\u0000.CalendarIntentService\u0000&\'\u0000\u0000\u0000n\u0005\u0000\u0000' }
{ date: Tue Jul 22 2014 10:27:58 GMT-0400 (EDT),
  pid: 797,
  tid: 1209,
  priority: 79,
  tag: 'u',
  message: '\u0000\u0003\u0003\u0000\u0000\u0000\u0000\u0000\u0000��\u001b\f\u0000n\u0005\u0000\u0000' }
{ date: Tue Jul 22 2014 10:27:58 GMT-0400 (EDT),
  pid: 24879,
  tid: 24879,
  priority: 4,
  tag: 'wpa_supplicant',
  message: 'wlan0: CTRL-EVENT-SCAN-STARTED ' }
```

Notice that they are surrounded by otherwise normal-appearing entries.

**Comment by sorccu** [comment]
It looks like these values are lists in a special binary format. For example, the following:

``` javascript
{
message: '\u0000\u0003\u0005\u0000\u0000\u0000\u0000\u0000\u0000!\u0004�\u0010\u0002\u000f\u0000\u0000\u0000.CheckinService\u0000\u0017\'\u0000\u0000\u0000�\u0005\u0000\u0000'
}
```

Might show up like this in the regular logcat format:

```
I/am_create_service( 1256): [0,284886049,.CheckinService,10007,1414]
```

So, we'll just have to add support for that. I've actually seen that format before in the "events" log stream (`adb logcat -b events`), but looks like L is using it more liberally.

**Comment by syncxplus** [comment]
Hi guys, I met this issue recently. 
I find one reason is that some devices using all the V1, V2, V3 versions of logger_entry from time to time. Here I have a quick fix and it works well on my environment. 
I will start the pull request to the develop branch for your reference.

**Comment by nwanglu** [comment]
@sorccu, Is this issue fixed? the version from npm still could not get correct log for android L.

**Comment by sorccu** [comment]
Well, that took a while, but released as 1.1.0 now.
